### ISSUE TITLE
List of Small Corrections - Docs

### ISSUE THREAD

**Issue by ramirezfranciscof** [aiidateam/aiida-core#3874]
This a general issue to gather small corrections to the documentation. Once there is a critical number of these or enough time has passed, these will be incorporated through a single PR.

Edit by SPH: I will use the rocket emoticon if the problem has already been taken care of during the `docs-revamp` or is obsolete. If it will be addressed by #4133 it'll get the thumbs up

**Comment by ramirezfranciscof** [comment]
"We have **know** defined through the process specification, what inputs the calculation job expects and what outputs it will create"

Source: https://aiida.readthedocs.io/projects/aiida-core/en/latest/working/calculations.html#prepare

**Comment by mbercx** [comment]
"Data provenance: Automatically track inputs, ~~outpus~~ **outputs** & metadata of all calculations in a provenance graph for full reproducibility. Perform fast queries on graphs containing millions of nodes."

Source: https://github.com/aiidateam/aiida-core/blob/develop/docs/source/index.rst#features

"For the advanced **user**, there is the complete :doc:`AiiDA API reference<apidoc/aiida>` (including a :ref:`modindex`), and of course you can always peek into the code on the AiiDA git repository."

Source: https://github.com/aiidateam/aiida-core/blob/develop/docs/source/index.rst#getting-started

**Comment by mbercx** [comment]
"This behavior is basically the result of the settings of call_calc_forward=True and call_work_forward=True, which makes that the inclusion of a process node will also imply the inclusion of any child or parent process node as well. Following ~~this~~ **these** rules in a recursive way leads to the command affecting all the processes within any given workflow: in this way, nodes that are sub-processes of a given highest-level workflow will end up grouped together, in the sense that (by default) they will all be affected in the same way when deleting or exporting."

Source: https://aiida-core.readthedocs.io/en/latest/concepts/provenance.html#cascading-rules-an-example

**Comment by mbercx** [comment]
- The `class` and `__init__` method definitions of the `NewData` code snippets are missing the colon `:` at the end. (Also in the snippet with the `@property` decorator.)

- "Before calling the ~~construtor~~ **constructor** of the base class, we have to remove the value keyword from the keyword arguments kwargs, because the base class will not expect it and will raise an exception if left in the keyword arguments."

- The code snippet where the `NewData` class is supplied with the `value` property no longer contains the class constructor. Should we not include this, possible using an ellipsis `...` to hide its body?

Source: https://aiida-core.readthedocs.io/en/latest/working/data.html#creating-new-data-types

"[...] Of course a data type may need to store multiple properties of varying character, but both storage locations can safely be used in ~~parellel~~ **parallel**. [...]"

Source: https://aiida-core.readthedocs.io/en/latest/working/data.html#database-or-repository

**Comment by mbercx** [comment]
"For basic information on the concept of a CalcJob or calcfunction, refer to the calculations concept ~~and the same for t~~. **T**he WorkChain and workfunction ~~is~~ **are** described in the workflows concept. [...]"

Source: https://aiida-core.readthedocs.io/en/latest/concepts/processes.html#process-types

- "The previous section [...]. When the process is being ‘~~ran~~ **run**’ (see the section on launching processes for more details) [...]. When a process is ‘submitted’, an instance of the Process is created, along with the node that represents it in the database, and its state is then __persisted__ _(I'm not sure what is meant here)_ to the database. [...]"

- "shutdown" should be replaced by "shut down" when used as a verb.

Source: https://aiida-core.readthedocs.io/en/latest/concepts/processes.html#process-tasks

The property `is_sealed` should not finish with brackets `()`.

Source: https://aiida-core.readthedocs.io/en/latest/concepts/processes.html#process-sealing

**Comment by sphuber** [comment]
[QueryBuilder append docs ordering](https://github.com/aiidateam/aiida-core/blob/develop/docs/source/querying/querybuilder/append.rst#ordering-results)
```
qb = QueryBuilder()
qb.append(
        CalcJobNode,
        project=['*']
    )
qb.append(
        Dict,
        filters={'attributes.energy':{'>':-5.0}},
     )

qb.order_by({CalcJobNode: [
    {'ctime': 'asc'},
    {'attributes.energy': {'order': 'desc', 'cast': 'f'}}  # 'f' is an alias for type float
}])
```
The terminating `}])` should be `]})`
Originally reported in #3907

**Comment by sphuber** [comment]
Replace `from aiida.backends.tests.utils.archives import get_archive_file` with `from tests.utils.archives import get_archive_file`. Originally reported in #3911

**Comment by mbercx** [comment]
"Often when designing the specification of a Process, we cannot know exactly which inputs we want to be able to pass to the process. However, with the concept of the InputPort and OutputPort one does need to know exactly, how many ~~value~~**values** one expects at least, as they do have to be defined. "

"For this situation, the InputPort has the attribute non_db. By default this is set to False, but by setting it to True **we can indicate** ~~the port is marked~~ that the values that are passed to ~~it~~ **the port** should not be stored as a node in the provenance graph and linked to the process node. [...]"

"This work chain can now be called with native Python types, which will automatically **be** converted to AiiDA types by the aiida.orm.nodes.data.base.to_aiida_type() function."

"This is useful, because the caller can now programmatically, based on the exit_status, decide how to proceed. This is an infinitely more robust way of ~~communcating~~**communicating** specific errors to a non-human ~~then~~**than** parsing text based **on** logs or reports. Additionally, ~~The~~**the** exit codes make it ~~also~~ very easy to query for failed processes with specific error codes."

"Each process, in addition to the normal inputs defined through its process ~~specifcation~~**specification**, [...]"

"As explained in a previous section, the inputs for a CalcJob and WorkChain are defined in the define() method. To know ~~then~~ what inputs they take, [...]"

"This command is most useful for WorkChain instances, but also works for CalcJobs. One of the more powerful ~~aspect~~**aspects** of work chains, [...]"

"[...]. By definition, these runners will always run in a different system process ~~then~~**than** the one from which you want to interact, [...]"

"Depending on the cause [...]. ~~If you think the The problem is, h~~**H**owever, there is unfortunately no way of telling what the actual problem is for the process not being reachable. [...]"

Source: https://aiida-core.readthedocs.io/en/latest/working/processes.html

**Comment by csadorf** [comment]
Are all the issues addressed in #4135?
